### PR TITLE
Update Pull Request checklist to guard better against bugs

### PR DESCRIPTION
**Pull request by alexsoble**
# Notes 

+ One of my big learnings from #1276 was that my QA on #1261 wasn't thorough enough. I tested a bunch of the site but not the Student Report, which went down because of a JS bug.

+ This PR makes the checklist feature way more specific. First of all, it asks the PR creator to name the parts of the app that are affected by any JS changes. That way we should be reminded on each PR to do targeted, specific QA before we merge.

+ In addition, it makes separate checklists for the author and for the reviewer. That's because the current checklist format is ambiguous about who should do the QA: the author, the reviewer, or both? I think both so that's how I wrote the checklist.

### CODE REVIEW

**Comment by alexsoble** [comment]
@gabivoicu @edavidsonsawyer @Jngai @jhilde Let me know what you think of this checklist proposal!

**Comment by Jngai** [comment]
@alexsoble I especially like the 2nd one. That would have caught the bug.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -8,4 +8,15 @@
...
 # Checklist
 
-+ [ ] Tested latest version in an Internet Explorer virtual machine? *(If the PR adds Javascript.)*
+*If the PR touches Javascript, follow this checklist. If the PR doesn't touch Javascript, you can delete this checklist.*
+
+*Which features or pages does this PR touch? Make a list here so that you can double-check each affected page in a Internet Explorer before merging. Make one list for the author and another for the reviewer. For more info on how to download and use a Virtual Machine to test in IE, see the README.*
+
+*Examples:*
+
++ [ ] Author checked latest in IE - Student Profile
```
Comment (kevinrobinson) [inline .github/PULL_REQUEST_TEMPLATE.md]: Can we write automated tests for any of this?  The bugs mentioned in the description don't have anything to do with IE and would have been caught by any tests that exercised that code.   In particular, I'm wondering if we could write a JS test for the student profile page that renders three data points for each chart, and then verifies there are that many data points on the page.

Also, checklists and manual testing are great too!  I'm just wondering if automated tests could help as well.

Comment (alexsoble) [reply .github/PULL_REQUEST_TEMPLATE.md]: Yes to more tests! And to checklists. How about an item in the checklist that talks about adding test coverage?

**Comment by gabivoicu** [comment]
I think making the checklist more specific is a fantastic idea, as is being more clear about whose responsibility it is to test. 👍  for me!

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (Jngai) [reply .github/PULL_REQUEST_TEMPLATE.md]: When I was working on the refactoring I also notice parts of the app that doesn't have specs for. I was considering adding them. Here is also a vote from me for more tests.

**Comment by alexsoble** [comment]
@Jngai @gabivoicu @kevinrobinson The people have spoken! :D I added two checklist items about automated test coverage here: 3682f72050432ccc77a53c03466db6edabb4d298. Feel free to add more suggestions and/or submit a patch to this PR with any language or ideas you'd like to add.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (Jngai) [reply .github/PULL_REQUEST_TEMPLATE.md]: @alexsoble @kevinrobinson Have you guys seen `yarn test --coverage`? Lots of js in the red and `app/assets/javascripts/student_profile/pdf` is simply not tested.

**Comment by alexsoble** [comment]
On a weekly call with @gabivoicu and @Jngai right now!

@Jngai is my reviewer, she's giving me her 👍  to merge.

**Pull request merged**